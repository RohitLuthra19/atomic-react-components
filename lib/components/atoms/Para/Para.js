// @flow
import React from 'react';
import type { Node } from 'react';
import styled from 'styled-components';

import styles from './Para.style';

type Props = {
  children: Node,
  className?: string | void,
  inheritedStyles?: string,
  elementLocator?: string,
};

const Para = ({ children, className, inheritedStyles, elementLocator, ...others }: Props): Node => {
  const dataLocator = elementLocator
    ? {
        'data-locator': `para-${elementLocator}`,
      }
    : {};

  return (
    <p className={className} {...dataLocator} {...others}>
      {children}
    </p>
  );
};

Para.defaultProps = {
  inheritedStyles: '',
  className: '',
  elementLocator: '',
};

export default styled(Para)`
  ${styles};
`;
export { Para as ParaVanilla };
