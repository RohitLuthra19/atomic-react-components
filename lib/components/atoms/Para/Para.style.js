import { css } from 'styled-components';

export default css`
  letter-spacing: 0px;
  font-weight: normal;
  font-family: ${props => props.theme.primaryFontFamily};
  font-size: 12px;
`;
