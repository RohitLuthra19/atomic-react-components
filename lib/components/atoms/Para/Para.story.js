// @flow
import React from 'react';
import { storiesOf } from '@storybook/react';

// Import Vanilla Component to display markup and props
import { ParaVanilla } from './Para';
// Import Styled Component to showcase variations
import Para from './index';

storiesOf('Atoms', module).addWithChapters('Para', {
  chapters: [
    {
      sections: [
        {
          sectionFn: () => <ParaVanilla className="hide-default-sample">This is plain text</ParaVanilla>,
          options: {
            showSource: true,
            allowSourceToggling: true,
            showPropTables: true,
            allowPropTablesToggling: true,
          },
        },
      ],
    },
    {
      title: 'Input Field Error Message',
      sections: [
        {
          title: 'Default',
          sectionFn: () => <Para>This is plain text.</Para>,
        },
      ],
    },
  ],
});
