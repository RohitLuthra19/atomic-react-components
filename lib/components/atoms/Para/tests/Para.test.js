import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import Para from '..';

const props = {
  children: 'Sample',
  className: '',
};

describe('Para Component', () => {
  let paragraph;

  test('should render correctly and match snapshot', () => {
    paragraph = shallow(<Para {...props}>This is Para text</Para>);
    expect(paragraph).toMatchSnapshot();
  });

  test('should render with styles correctly and match snapshot', () => {
    paragraph = renderer.create(<Para {...props}>This is Para text</Para>).toJSON();
    expect(paragraph).toMatchSnapshot();
  });
});
