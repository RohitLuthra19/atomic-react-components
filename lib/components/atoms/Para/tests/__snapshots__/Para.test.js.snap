// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Heading Component should render correctly and match snapshot 1`] = `
.c0 {
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  font-weight: normal;
  font-size: 20px;
}

<Para
  className="c0"
  inheritedStyles=""
  type="h3"
>
  Sample
</Para>
`;

exports[`Heading Component should render with styles correctly and match snapshot 1`] = `
.c0 {
  -webkit-letter-spacing: 1px;
  -moz-letter-spacing: 1px;
  -ms-letter-spacing: 1px;
  letter-spacing: 1px;
  font-weight: normal;
  font-size: 20px;
}

<p
  className="c0"
>
  Sample
</p>
`;
